### ISSUE TITLE
Show first name & last name in connections list on home page

### ISSUE THREAD

**Issue by akhilchoudhary2k** [akhilchoudhary2k/MindMatch-kgp#34]
see image for reference.
<img width="860" alt="Screenshot 2022-10-27 at 1 58 04 AM" src="https://user-images.githubusercontent.com/42896296/198130351-39ef5ff4-57af-44ff-b7ed-95b8979f7e58.png">

**Comment by RandomUserWithInternet** [comment]
Could I take this issue?

**Comment by akhilchoudhary2k** [comment]
Yes pls go ahead @RandomUserWithInternet , but try to raise PR ASAP other wise I may reassign it.
As it is an easy fix.

**Comment by RandomUserWithInternet** [comment]
Trying to access the fname and lname data by looking for other users via connections, but I don't know where user data of everyone else on the site is located.

**Comment by akhilchoudhary2k** [comment]
in `app.get("/UserHome"` we do `res.render('UserHome', {user: req.user});`
then in `UserHome.ejs` we get the name of ith connection from `user.connections[i]`

it means we don't pass the fname & lname information, `user.connections[i]` stores only the username.
maybe you can pass a maping of username --> {fname, lname} or pass 2 lists fname_list & lname_list.

@RandomUserWithInternet 
in short: the implementation depends on you, but you need to query the DB to get fname & lnames then pass it, that's it.

**Comment by RandomUserWithInternet** [comment]
I don't think I'm familiar with the DB syntax enough to solve this issue. I understand that only the username is available, but I'm just not sure how to use it to look up in the database the fname/lname it's connected to and pull the necessary information. Sorry about that.

**Comment by RandomUserWithInternet** [comment]
Pretty new to GitHub myself, so IDK if there's a way to unassign myself or not. I'll let you deal with it ig lol.

**Comment by akhilchoudhary2k** [comment]
@RandomUserWithInternet Will you be able to do it if I create a function to get fname & lname from username which does the DB query ??

roughly it will be like : 
```
User.findOne({
                username: username
            }, function(err, found) {
            if(found) {
                   return found.fname, found.lname
            }
});
```

**Comment by RandomUserWithInternet** [comment]
Yeah I'll try it out.

**Comment by RandomUserWithInternet** [comment]
![image](https://user-images.githubusercontent.com/88126013/198145352-ab18821d-6c27-49c0-960f-d11d147984e1.png)
![image](https://user-images.githubusercontent.com/88126013/198145536-5adebdaa-0386-4c79-b7e4-6850f1b10873.png)
Here's what I did... Not sure if I implemented it correctly.

**Comment by akhilchoudhary2k** [comment]
you are on the right lines, but it's not correct.
will help you fix it tomorrow.

**Comment by RandomUserWithInternet** [comment]
Thanks! I appreciate you helping me out, in spite of my incompetence.

**Comment by RandomUserWithInternet** [comment]
Just taking a second look, I'm assuming I need to return the User.findOne function as well. 
Also, what will it return if it is empty? Would it be null or something else?

**Comment by akhilchoudhary2k** [comment]
@RandomUserWithInternet 
--> it will return null if empty, so replace that with "NA"

--> create a list `connectionsFullName` and iterate on connections and append the [fname, lname] into this `connectionsFullName` list and pass it to res.render()

--> this `getNames` function may not run in order because it contains a time costly DB operation, you need to take care of synchronization.  link for reference: [LINK](https://sebhastian.com/javascript-wait-for-function-to-finish/)

**Comment by RandomUserWithInternet** [comment]
Alright, I'll try and implement those things. Could you explain to me what res.render() is? I see it elsewhere in the app.js file but don't know what exactly it does.

**Comment by akhilchoudhary2k** [comment]
Refer to internet and documentation.
But in a nut shell `res.render( x, {par1:val1, par2:val2......} )` will send/show/render `x.ejs` page to the client and pass an object of key-value pairs which will be displayed in x.ejs page.

**Comment by RandomUserWithInternet** [comment]
I haven't fixed synchronization yet, working on it now. Just checking to see if I have it right so far. 

app.js:
```js
function getFullNames(){
    let connectionFullName = {};
    for (let i in user.connections){
        var username = user.connections[i];
        var fullName = User.findOne({
            username: username
        }, function(err, found) {
            if(found) {
            return [found.fname, found.lname]
            }
        });

        if (fullName[0] == null){
            fullName[0] = "NA";
        }
        if (fullName[1] == null){
            fullname[1] = "NA";
        }

        connectionFullName[user.connections[i]] = fullName;
    }

    res.render('UserHome', connectionFullName);
}
```
UserHome.ejs:
![image](https://user-images.githubusercontent.com/88126013/198368545-e0afca9b-8b74-4072-87d2-2e399d15f525.png)
![image](https://user-images.githubusercontent.com/88126013/198367674-79d815df-75d4-4f89-9643-b5beeb02568c.png)

**Comment by RandomUserWithInternet** [comment]
Hopefully that's a correct implementation of the promise... 

![image](https://user-images.githubusercontent.com/88126013/198373134-9258d4fa-c6f8-4332-a5cc-cdda1d32a76c.png)

**Comment by akhilchoudhary2k** [comment]
Run the code and check

**Comment by RandomUserWithInternet** [comment]
Is there a test account I can log in with? It crashed when I tried to register.

**Comment by RandomUserWithInternet** [comment]
The errors seem to do with promises, so not so sure anymore. Let me try to run the base version.

**Comment by RandomUserWithInternet** [comment]
Has the same error.

**Comment by RandomUserWithInternet** [comment]
I hope I'm not doxing myself. 
![image](https://user-images.githubusercontent.com/88126013/198382094-a0a03ca1-a26e-4e21-b013-4e0c51048c6c.png)

**Comment by akhilchoudhary2k** [comment]
Because you have not setup the database correctly.
I told you to go through readme file and steps on how to run the code locally.

**Comment by RandomUserWithInternet** [comment]
Followed the README from the beginning. Had to install the plugins to lol. I had problems with the mongo commands from the terminal, so I used the Compass GUI instead. I don't know how that would affect anything though... I understand you can't help help everyone debug everything, however, I don't think I'll be able to test run it on my own. If someone else could test run my code, maybe that could work. I'll just make a pull request and label it as a draft.
